### PR TITLE
Add authorization and saving to Neo4j

### PR DESCRIPTION
**Pull request by spisladqo**
Implement functions to save graph to and load it from Neo4j.

Add window for authorization in Neo4j AuraDB. It appears if user hasn't logged in, while trying to save graph to Neo4j.

Add saving graph to Neo4j through GUI (but not loading from it _yet_).

Add dialog window on load, which shows all stored graphs.

Add error messages describing what went wrong when working with Neo4j.
 
Make gradle cleaner.

### CODE REVIEW

**Review by shakareem — APPROVED** [review]

File: app/src/main/kotlin/view/utils/Neo4jLoginDialog.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import androidx.compose.ui.layout.HorizontalAlignmentLine
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import kotlinx.coroutines.delay
+import viewmodel.MainScreenViewModel
+
+@Composable
+fun <D> Neo4jLoginDialog(
+    mainScreenViewModel: MainScreenViewModel<D>,
```
Comment (qrutyy) [inline app/src/main/kotlin/view/utils/Neo4jLoginDialog.kt]: Why do you need MSVM?

File: app/src/main/kotlin/viewmodel/MainScreenViewModel.kt [hunk 2]
```diff
@@ -14,10 +18,48 @@ class MainScreenViewModel<D>(
...
+        ?: GraphViewModel(
+            graph,
+            showVerticesIds,
+            showVerticesData,
+            graphType,
+            setDirectionState(currentGraphType),
+            setWeightinessState(currentGraphType)
+        )
+
+    fun tryInitNeo4jRepo(uri: String, user: String, password: String): Boolean {
```
Comment (qrutyy) [inline app/src/main/kotlin/viewmodel/MainScreenViewModel.kt]: Idk if it suits well here.

File: app/src/main/kotlin/view/tabScreen/GeneralTab.kt [hunk 3]
```diff
@@ -63,10 +63,7 @@ fun <D> GeneralTab(graphVM: GraphViewModel<D>) {
                     onClick = { if (vertexData.isNotEmpty()) showVertexAddDialog = true },
                     colors = ButtonDefaults.buttonColors(MaterialTheme.colors.primary)
                 ) {
-                    Text(
-                        "add\nvertex",
-                        textAlign = TextAlign.Center
-                    )
+                    Text("add")
```
Comment (qrutyy) [inline app/src/main/kotlin/view/tabScreen/GeneralTab.kt:66]: @kar1mgh just edited these lines, please be careful with solving conflicts!

**Review by qrutyy — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/viewmodel/MainScreenViewModel.kt (hunk 2 above)
Comment (spisladqo) [reply app/src/main/kotlin/viewmodel/MainScreenViewModel.kt]: Removed

File: app/src/main/kotlin/view/utils/Neo4jLoginDialog.kt (hunk 1 above)
Comment (spisladqo) [reply app/src/main/kotlin/view/utils/Neo4jLoginDialog.kt]: I looked over my decisions and I agree with you

**Review by qrutyy — APPROVED** [review]
GJ!

**Pull request merged**